docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_distributions_https.mdx

```diff
@@ -23,6 +23,262 @@ HITRUST, SOC2, NISTCSF, PCIDSS
+
+   ```sh
+   aws cloudfront update-distribution --id <DISTRIBUTION_ID> --distribution-config file://distribution-config.json
+   ```
+
+   In your `distribution-config.json`, include the `ACMCertificateArn`:
+
+   ```json
+   {
+     "ViewerCertificate": {
+       "ACMCertificateArn": "arn:aws:acm:region:account-id:certificate/certificate-id",
+       "SSLSupportMethod": "sni-only",
+       "MinimumProtocolVersion": "TLSv1.2_2018"
+     }
+   }
+   ```
+
+3. **Ensure All Cache Behaviors Use HTTPS:**
+   Make sure all cache behaviors in your distribution are set to use HTTPS.
+
+   ```sh
+   aws cloudfront update-distribution --id <DISTRIBUTION_ID> --distribution-config file://distribution-config.json
+   ```
+
+   In your `distribution-config.json`, for each cache behavior, set `ViewerProtocolPolicy` to `redirect-to-https` or `https-only`:
+
+   ```json
+   {
+     "CacheBehaviors": [
+       {
+         "PathPattern": "/path",
+         "ViewerProtocolPolicy": "redirect-to-https"
+       }
+     ]
+   }
+   ```
+
+4. **Verify Distribution Configuration:**
+   After making changes, verify the configuration to ensure HTTPS is enabled.
+
+   ```sh
+   aws cloudfront get-distribution-config --id <DISTRIBUTION_ID>
+   ```
+
+   Check the output to confirm that `ViewerProtocolPolicy` is set to `redirect-to-https` or `https-only` and that the `ViewerCertificate` is properly configured.
+
+By following these steps, you can ensure that HTTPS is enabled on your CloudFront distributions using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not having HTTPS enabled on CloudFront distributions using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK for Python (Boto3):**
+   - Ensure you have Boto3 installed. If not, install it using pip:
+     ```bash
+     pip install boto3
+     ```
+
+2. **Create a Boto3 Session:**
+   - Initialize a session with your AWS credentials.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   ```
+
+3. **Retrieve and Update CloudFront Distribution Configurations:**
+   - Fetch the current configuration of your CloudFront distribution.
+   - Modify the configuration to ensure HTTPS is enabled.
+   ```python
+   client = session.client('cloudfront')
+
+   distribution_id = 'YOUR_DISTRIBUTION_ID'
+
+   # Get the current distribution configuration
+   response = client.get_distribution_config(Id=distribution_id)
+   config = response['DistributionConfig']
+   etag = response['ETag']
+
+   # Ensure HTTPS is enabled
+   for origin in config['Origins']['Items']:
+       if 'CustomOriginConfig' in origin:
+           origin['CustomOriginConfig']['OriginProtocolPolicy'] = 'https-only'
+       elif 'S3OriginConfig' in origin:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>